gobblin-runtime/src/main/java/org/apache/gobblin/runtime/api/MutableSpecCatalog.java

```diff
@@ -22,27 +22,27 @@
 import java.util.Properties;
 import java.util.concurrent.TimeUnit;
 
-import org.apache.gobblin.instrumented.Instrumented;
-import org.apache.gobblin.metrics.ContextAwareTimer;
-
 import com.google.common.base.Optional;
 import com.typesafe.config.Config;
 
 import lombok.Getter;
 import lombok.extern.slf4j.Slf4j;
 
+import org.apache.gobblin.instrumented.Instrumented;
+import org.apache.gobblin.metrics.ContextAwareTimer;
+
 
 /**
  * A {@link SpecCatalog} that can have its {@link Collection} of {@link Spec}s modified
  * programmatically. Note that specs in a spec catalog can change from the outside. This is covered
  * by the base SpecCatalog interface.
  */
-public interface MutableSpecCatalog extends SpecCatalog {
+public interface MutableSpecCatalog<T> extends SpecCatalog {
```

Comment:
I think we could just directly use Map<String, AddSpecResponse> as our return type in put() API, so that T generic is not needed here. The topology catalog could simply put an empty AddSpecResponse if it doesn't have anything to return.

Reply:
Made the change.